src/content/language/pattern-types.md

```diff
@@ -288,11 +288,10 @@ x || y && z => 'matches true',
 // ...
 ```
 
-In the first case, the logical-and pattern `y && z` evaluates first because
-logical-and patterns have higher precedence than logical-or.
-In the next case, the logical-or pattern is parenthesized. It evaluates first,
-which results in a different match.
-
+Dart evaluates `x || y && z` as `x || (y && z)`,
```

Comment:
It is again dangerous to say this, because Dart doesn't evaluate a pattern: It performs matching operations that may succeed or fail (or throw), and the resulting behavior depends on whether or not there was a match for the entire pattern.

What we _can_ say is that `p1 || p2` proceeds by matching the matched value with `p1` and succeeds if the match succeeded, otherwise it matches the matched value with `p2` and succeeds if the match succeeded; and otherwise the match fails.

I think it's safe to say that the pattern `x || y && z` is equivalent to the pattern `x || (y && z)`, and then we can use the description of how to match an object against `x` respectively how to match that object against `y && z` to understand how the pattern `x || (y && z)` behaves.

Reply:
I think it might actually be a really good point to make that `(x || y) && z` doesn't match anything, because this puts a strong emphasis on the fact that pattern matching isn't boolean expression evaluation.